datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -374,15 +519,29 @@ impl RecordBatchStream for GroupedHashAggregateStream {
 
 impl GroupedHashAggregateStream {
     /// Perform group-by aggregation for the given [`RecordBatch`].
-    fn group_aggregate_batch(&mut self, batch: RecordBatch) -> Result<()> {
+    fn group_aggregate_batch(&mut self, batch: RecordBatch, merging: bool) -> Result<()> {
         // Evaluate the grouping expressions
-        let group_by_values = evaluate_group_by(&self.group_by, &batch)?;
+        let group_by_values = if merging {
+            let group_by = PhysicalGroupBy::new_single(self.group_by.expr.clone());
```

Comment:
I wonder if we need to create this PhysicalGroupBy for each batch? Why not create it once at the start?

Reply:
Updated